servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AlpnChannelSingle.java

```diff
@@ -93,17 +97,34 @@ protected void configurePipeline(final ChannelHandlerContext ctx, final String p
         @Override
         protected void handshakeFailure(final ChannelHandlerContext ctx, final Throwable cause) {
             LOGGER.warn("{} TLS handshake failed:", ctx.channel(), cause);
-            failSubscriber(cause, ctx.channel());
+            if (!failSubscriber(cause, ctx.channel())) {
+                ChannelCloseUtils.close(ctx, cause);
+            }
         }
 
         @Override
         public void exceptionCaught(final ChannelHandlerContext ctx, final Throwable cause) {
+            Throwable wrapped;
+            // This unwrapping logic is copied from the parent ApplicationProtocolNegotiationHandler
+            if (cause instanceof DecoderException && ((wrapped = cause.getCause()) instanceof SSLException)) {
```

Comment:
Should the handshakeFailure also be triggered if a different exception is raised?

Reply:
After we moved observer notification logic to `SslHandler.handshakeFuture()`, it doesn't matter anymore. Observer will be notified anyway.
For the pipeline, any other exception will either be propagated to the subscribed (users will see it) or channel will be closed (users already got some other exception before). 
Therefore, choice is to make it as consistent as possible to its parent implementation.